### PR TITLE
Auto-review required dependency matching in composer.json

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
This introduces "auto-review tests". Automate the boring task of self-reviewing your own code through GHA.

In this first iteration, this tests the composer.json in root against in `admin/framework` to check if the "require" sections are the same.

This needs #5531 to go green.

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: .github/workflows/test-phpunit.yml [hunk 1]
```diff
@@ -141,9 +141,45 @@ jobs:
...
           COVERALLS_FLAG_NAME: PHP ${{ matrix.php-versions }} - ${{ matrix.db-platforms }}
 
+  auto-review-tests:
+    name: Automatic Code Review
+    needs: [unit-tests]
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout
+        uses: actions/cache@v2
```
Comment (kenjis) [inline .github/workflows/test-phpunit.yml:151]:
```suggestion
        uses: actions/checkout@v2
```

Comment (paulbalandan) [reply .github/workflows/test-phpunit.yml]: Thanks!

**Comment by kenjis** [comment]
Good.

```
There was 1 failure:

1) CodeIgniter\AutoReview\ComposerJsonTest::testFrameworkRequireIsTheSameWithDevRequire
The framework's "require" section is not updated with the main composer.json.
Failed asserting that two arrays are identical.
--- Expected
+++ Actual
@@ @@
     'ext-intl' => '*'
     'ext-json' => '*'
     'ext-mbstring' => '*'
-    'kint-php/kint' => '^4.0'
+    'kint-php/kint' => '^3.3'
     'laminas/laminas-escaper' => '^2.9'
     'psr/log' => '^1.1'
 )

/home/runner/work/CodeIgniter4/CodeIgniter4/tests/AutoReview/ComposerJsonTest.php:35
```

**Review by kenjis — APPROVED** [review]

**Review by MGatner — COMMENTED** [review]
This is great! I'll think about some other things that might fit here. 

Is there a compelling reason to group this with the existing Action? I'd be in favor of splitting it out, especially since it covers nothing.

**Comment by kenjis** [comment]
Agree with @MGatner. Making another `.yml` seems to be better.

And rebase to pass GA, please.

**Review by MGatner — APPROVED** [review]
Looks great! Thanks @paulbalandan

**Pull request merged**